docs/configuration.md

```diff
@@ -0,0 +1,165 @@
+# Configuring Spectral
```

Comment:
Can we make it clear that this is about writing ruleset and not config? That’s a different thing to be done later.

Reply:
My intention was to store all that stuff in this file.
Setting custom rulesets _is_ imo configuration. On top of that there will be other stuff like config file etc. I tried to split this file into sections to make that clear. Is it not?